src/components/Api.js

```diff
@@ -0,0 +1,101 @@
+export default class Api {
+  //***API CONSTRUCTOR***
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  //***RENDER RESULT FROM SERVER***
+  renderResult(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error :${res.status}`);
+    }
+  }
+  /*/***METHOD TO CHECK RESPONSE FROM SERVER***
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error :${res.status}`);
+  }
+
+  //*** REQUEST RESPONSE FROM SERVER***
+  async _request(url, options) {
+    const res = await fetch(url, options);
+    return this._checkResponse(res);
+  }*/
+
+  //***TO GET USER INFO METHOD***
+  async getUserInfo() {
+    const res = await fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    });
+    return this.renderResult(res);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this.renderResult)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.